staging/src/k8s.io/apiserver/pkg/storage/cacher/lister_watcher_test.go

```diff
@@ -119,3 +125,52 @@ func TestCacherListerWatcherPagination(t *testing.T) {
 	}
 
 }
+
+func TestCacherListerWatcherListWatch(t *testing.T) {
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.WatchList, true)
+
+	prefix := "pods"
+	fn := func() runtime.Object { return &example.PodList{} }
+	server, store := newEtcdTestStorage(t, prefix)
+	defer server.Terminate(t)
+
+	makePodFn := func() *example.Pod {
+		return &example.Pod{
+			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "test-ns"},
+		}
+	}
+	createdPod := &example.Pod{}
+	key := computePodKey(createdPod)
```

Comment:
The key is now computed from the empty pod - this seems wrong.
Lets do sth like:
```
pod := makePodFn()
key := computePodKey(pod)
createdPod := &example.Pod{}
...
```